tests/typescript-mock-data.spec.ts

```diff
@@ -51,36 +60,61 @@ it('should generate mock data functions with external types file import', async
     const result = await plugin(testSchema, [], { typesFile: './types/graphql.ts' });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
+    expect(result).toContain(
+        "import { ABCType, Avatar, UpdateUserInput, User, ABCStatus, Status } from './types/graphql';",
+    );
     expect(result).toMatchSnapshot();
 });
 
 it('should generate mock data with typename if addTypename is true', async () => {
     const result = await plugin(testSchema, [], { typesFile: './types/graphql.ts', addTypename: true });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
     expect(result).toMatchSnapshot();
 });
 
 it('should generate mock data with pascalCase enum if enumValues is "pascal-case#pascalCase"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'pascal-case#pascalCase' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with upperCase enum if enumValues is "upper-case#upperCase"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'upper-case#upperCase' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with as-is enum if enumValues is "keep"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'keep' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with pascalCase types by default', async () => {
+    const result = await plugin(testSchema, [], {});
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with upperCase types if typenames is "upper-case#upperCase"', async () => {
     const result = await plugin(testSchema, [], {
-        enumValues: 'pascal-case#pascalCase',
-        typesFile: './types/graphql.ts',
+        typenames: 'upper-case#upperCase',
     });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
     expect(result).toMatchSnapshot();
 });
 
-it('should generate mock data with upperCase enum if enumValues is "upper-case#upperCase"', async () => {
+it('should generate mock data with as-is types if typenames is "keep"', async () => {
     const result = await plugin(testSchema, [], {
-        enumValues: 'upper-case#upperCase',
-        typesFile: './types/graphql.ts',
+        typenames: 'keep',
     });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
```

Comment:
I agree however these expect where not the relevant ones, it should have been something like:
`expect(result).toContain('anABCTYPE')`

Reply:
Yes. I removed the one from the typename test and replaced it with a check for the `__typename` keyword.

I have added clearer matchers for all the case change tests and also added tests for default behavior vs pascalCase explicitly specified.